client/src/visualiser-src/sorting-visualiser/animation-producer/SortsBubbleAnimationProducer.ts

```diff
@@ -0,0 +1,53 @@
+import { boxWidth, whitespace } from '../util/constants';
+import { bubbleCodeSnippet } from '../util/codeSnippets';
+import GraphicalSortsElement from '../data-structure/GraphicalSortsElement';
+import SortsAnimationProducer from './SortsAnimationProducer';
+
+export default class SortsBubbleAnimationProducer extends SortsAnimationProducer {
+  public renderBubbleCode() {
+    this.renderCode(bubbleCodeSnippet);
+  }
+
+  public swap(
+    from: GraphicalSortsElement,
+    fromIndex: number,
+    to: GraphicalSortsElement,
+    isLast: boolean
+  ) {
+    const xFrom = fromIndex * (boxWidth + whitespace) + whitespace;
+    const cxFrom = fromIndex * (boxWidth + whitespace) + (boxWidth / 2 + whitespace);
+    const xTo = (fromIndex + 1) * (boxWidth + whitespace) + whitespace;
```

Comment:
Repetitive calculations here.
Create a function that takes in an index and returns an x value. 
Also, create another function that takes in a index and returns a cx value.
Put these in a file under util/helpers.ts.